### PR TITLE
feat: Add relabelings config to ServiceMonitor resource

### PR DESCRIPTION
**Pull request by thefirstofthe300**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Prometheus service discovery meta-labels `__meta_kubernetes_*` labels are not necessarily still attached to a metric when the metric_relabel_configs step is reached. Consequently, it is not possible to configure a label on pod as a metric tag using the chart solely. Adding the `relabelings` value to the ServiceMonitor will allow this use case. 

#### Does this PR introduce a user-facing change?

```release-note
Added relabelings config for the ServiceMonitor to the Helm chart
```

### CODE REVIEW

File: charts/cluster-autoscaler/Chart.yaml
Comment (Shubham82) [inline charts/cluster-autoscaler/Chart.yaml:14]: Hi @thefirstofthe300
This is a new feature, please make this a minor version bump.

Comment (thefirstofthe300) [reply charts/cluster-autoscaler/Chart.yaml]: Fixed.

**Comment by thefirstofthe300** [comment]
@gjtempleton @mwielgus Would you PTAL?

**Comment by Shubham82** [comment]
Hi @thefirstofthe300, Please address the failed test, which is required to pass to merge the PR.
It failed because you didn't bump the Chart version, so you need to bump the version to pass the failed test.

**Comment by Shubham82** [comment]
Here is the `Lint and Test Charts / Helm chart (pull_request)` test case failure error message:

```
The Lint and Test Charts / Helm chart (pull_request) test case failed due to not doing a bump version:
✖︎ cluster-autoscaler => (version: "9.34.1", path: "charts/cluster-autoscaler") > Chart version not ok. Needs a version bump! 
```
so you need to bump the chart version.

**Comment by Shubham82** [comment]
To update the README.md file, you have to update the README.md.gotmpl file and then with the help of pre-commit, it will update the README.md file

JFYI: you have to install [Pre-commit hooks](https://github.com/kubernetes/autoscaler/tree/master/charts#install-pre-commit-binary) before working in the Charts.

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by Shubham82** [comment]
@thefirstofthe300, please resolve the merge conflict.

**Comment by thefirstofthe300** [comment]
@gjtempleton @Shubham82 @jackfrancis Can I _please_ get an approval here? It's very frustrating to have what I judge to be a very simple PR languish in approval purgatory for five months. I appreciate Jack and Shubham taking the time to LGTM my PR, but not being able to merge despite those reviews is very frustrating.

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by jackfrancis** [comment]
@thefirstofthe300 fair criticism, I'll work to start reinforcing the helm chart (tests, auditing of features) to enable additional approvers in the near future

/assign @gjtempleton

**Comment by Shubham82** [comment]
It looks good to me.
/lgtm

**Pull request merged**
